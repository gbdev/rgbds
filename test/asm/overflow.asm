SECTION "sec", ROM0

MACRO print_x
	println x
ENDM

x = 2147483647
x = x + 1
	dl 2147483647+1
	print_x

x = -2147483648
x = x - 1
	dl -2147483648-1
	print_x

x = -2147483648
x = x * -1
	dl -2147483648 * -1
	print_x

x = -2147483648
x = x / -1
	dl -2147483648 / -1
	print_x

x = -2147483648
x = x % -1
	dl -2147483648 % -1
	print_x

x = -1
x = x << 1
	dl -1 << 1
	print_x

x = 4294967295
x = 4294967296

x = `33333333
x = `333333333
