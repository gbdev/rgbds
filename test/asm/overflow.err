warning: overflow.asm(23): [-Wdiv]
    Division of -2147483648 by -1 yields -2147483648
warning: overflow.asm(24): [-Wdiv]
    Division of -2147483648 by -1 yields -2147483648
warning: overflow.asm(44): [-Wlarge-constant]
    Integer constant is too large
warning: overflow.asm(47): [-Wlarge-constant]
    Graphics constant is too long, only first 8 pixels considered
