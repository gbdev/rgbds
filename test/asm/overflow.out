$80000000
$7FFFFFFF
$80000000
$80000000
$0
$FFFFFFFE
$80000000
$6B90BE55
